### ISSUE TITLE
Multiple proto files

### ISSUE THREAD

**Issue by cristaloleg** [getinsomnia/insomnia#2955]
**Is your feature request related to a problem? Please describe.**
Currently Insomnia doesn't support multiple *.proto files for gRPC. Which breaks our flow 'cause we've multiple files (for services, for events and for models). As you might guess some models(messages in terms of gRPC/protobuf) are not in the same file as services.

**Describe the solution you'd like**
Having an ability to select few files will resolve our problem. Looks like multiselect during request creation is more than enough (and ability to use gRPC methods from any file of course).

**Describe alternatives you've considered**
Still using autogenerated clients from terminal or other UI-based alternatives.

**Additional context**
nothing to add.

**Comment by develohpanda** [comment]
Hi @cristaloleg! It's interesting that you mention multi-select, we can definitely explore that as an avenue in the future.

We have a task in place for early Jan to work through supporting a root proto pattern for multiple proto files, where the main proto file `root.proto` imports every other file used by the project. I have a PoC that works in Insomnia, and just need to build some UI on top of it. 🙂 

Would this be suitable with your workflow?

**Comment by cristaloleg** [comment]
Yep, sounds exactly what we would like to see!

Our `proto` directory looks like this:

```
❯ exa --tree protos 
protos
├── events.proto
├── model.proto
└── svc.proto
```

**Comment by develohpanda** [comment]
Great! Planned for the next release in early 2021 😊

**Comment by loganintech** [comment]
Just to be 100% clear, does this require a single "root" proto, where all of the rest are in that tree?

We have a use case where we have every proto in a single parent folder, but not all of the protos are necessarily in the same import tree.

**Comment by cristaloleg** [comment]
Hi @loganintech can you show an example? tree structure or so

**Comment by loganintech** [comment]
Sure, here's a subset. Basically, the "connection" protos will include models, but aren't related to eachother.

```
proto
├── atoplevel
│  ├── connection
│  │  ├── service1
│  │  │  └── service1.proto
│  │  └── service2
│  │     └── service2.proto
│  ├── models
│  │  ├── service1
│  │  │  └── service1.proto
│  │  ├── service2
│  │  │  └── service2.proto
```

**Comment by develohpanda** [comment]
@loganintech thanks for that, the overall structure is not finalised yet so I'll keep this in mind when we start work on it!

**Comment by develohpanda** [comment]
Hey folks, please see screenshots in the linked PR for how this will behave in the first iteration. All of the patterns discussed in this issue will work. 🤗 Looking forward to your feedback once this is released!

**Comment by develohpanda** [comment]
Keeping this issue open for visibility, until the feature is released.

**Comment by chaitan94** [comment]
Excited to see this feature coming towards a release, it is looking great in the PR! Thanks for all the work @develohpanda :smile: :raised_hands:

**Comment by cristaloleg** [comment]
Looks reaaaaaally awesome, thank you! <3

**Comment by victor-pogor** [comment]
Hello. Thank you for implementing this feature. When will the new release be out?

**Comment by nijikokun** [comment]
> Hello. Thank you for implementing this feature. When will the new release be out?

Next release is slated for Early March, currently finalizing the next release and lots to still QA!

**Comment by dpclark4** [comment]
Hi, when is this expected to be released? It doesn't appear to be working in the latest beta, 2021.3.0-beta.1

**Comment by develohpanda** [comment]
Hi @dpclark4! This was released with 2021.1.0. In order to upload multiple protofiles you need to add a parent directory, see docs here: https://support.insomnia.rest/article/188-grpc#add-pf. I just confirmed on the latest beta and it seems to be working fine. 😄

**Comment by develohpanda** [comment]
Closing as released to stable in 2021.1.0!
